lib/tapioca/gem_info.rb

```diff
@@ -9,7 +9,7 @@ class GemInfo < T::Struct
     class << self
       extend(T::Sig)
 
-      #: (Bundler::LazySpecification spec) -> GemInfo
+      #: (::Bundler::StubSpecification | ::Gem::Specification spec) -> GemInfo
```

Comment:
I think you can just use `Gem::Specification`, it's a common ancestor


```suggestion
      #: (::Gem::Specification spec) -> GemInfo
```

```ruby
irb(main):001> require "bundler"
irb(main):002> Gem::Specification.ancestors
=>
[Gem::Specification::AllowSettingSource,
 Gem::Specification,
 Bundler::MatchPlatform,
 Bundler::GemHelpers,
 Bundler::MatchMetadata,
 Gem::BasicSpecification,
 Object,
 JSON::Ext::Generator::GeneratorMethods::Object,
 PP::ObjectMixin,
 Kernel,
 BasicObject]
irb(main):003> Gem::Specification.ancestors
=>
[Gem::Specification::AllowSettingSource,
 Gem::Specification,
 Bundler::MatchPlatform,
 Bundler::GemHelpers,
 Bundler::MatchMetadata,
 Gem::BasicSpecification,
 Object,
 JSON::Ext::Generator::GeneratorMethods::Object,
 PP::ObjectMixin,
 Kernel,
 BasicObject]
```


```suggestion
      #: (::Gem::Specification spec) -> GemInfo
```

Reply:
Huh, weird:

```
# It's using the normal Kernel#is_a?
Bundler.definition.specs.find(Bundler::StubSpecification).first.method(:is_a?)
=> #<Method: Bundler::StubSpecification(Kernel)#is_a?(_)>

Bundler.definition.specs.find(Bundler::StubSpecification).first.is_a?(Gem::Specification)
=> false # ? ? ?
```

Reply:
Oh, In my first snippet I print the `Gem::Specification.ancestors` twice 🤦‍♂️ 

Indeed, they're not compatible

```ruby
irb(main):002> Bundler::StubSpecification.ancestors
=>
[Bundler::StubSpecification,
 Bundler::RemoteSpecification,
 Comparable,
 Bundler::MatchPlatform,
 Bundler::GemHelpers,
 Bundler::FetchMetadata,
 Bundler::MatchRemoteMetadata,
 Bundler::MatchMetadata,
 Object,
 JSON::Ext::Generator::GeneratorMethods::Object,
 PP::ObjectMixin,
 Kernel,
 BasicObject]
```